### PR TITLE
Remove version and patch prompts from `omero db script`

### PR DESCRIPTION
**Pull request by manics**
- `omero db script`: only prompts for the root password
- `omero db script --password ome`: Shouldn't prompt for anything
- `omero db script --version DBVERSION --patch DBPATCH`: Set db.version and db.patch for developers
- `omero db script DBVERSION DBPATCH`: Old deprecated behaviour (positional args)
- All other existing flags should continue to work

https://trello.com/c/8PPlaBmP/456-remove-compulsory-version-patch-args-from-db-script
--no-rebase

### CODE REVIEW

File: components/tools/OmeroPy/test/unit/clitest/test_db.py [hunk 1]
```diff
@@ -136,14 +136,19 @@ def testScript(self, pos_input, no_salt, file_arg):
             output = self.file
         else:
             output = self.script_file
-        if "version" not in pos_input or "patch" not in pos_input:
-            self.expectVersion(self.data["version"])
-            self.expectPatch(self.data["patch"])
```
Comment (sbesson) [inline components/tools/OmeroPy/test/unit/clitest/test_db.py]: I suppose the two methods above are now obsolete and can be removed from the test class.

**Comment by sbesson** [comment]
Apart from my nitpicky suggestion, code changes look good and all unit tests are passing.

File: components/tools/OmeroPy/test/unit/clitest/test_db.py (hunk 1 above)
Comment (manics) [inline components/tools/OmeroPy/test/unit/clitest/test_db.py]: Removed

**Comment by sbesson** [comment]
Tested with ``OMERO.server-5.1.0-m5-503-5bc33e5-ice35-b444``. All worked as expected. Unit tests are green and expand the coverage. Thanks @manics .

**Pull request merged**